### DISCUSSION TITLE
Tasmota on ESP-01s get digital input on gpio 2

### DISCUSSION THREAD

**Discussion by tecbeat** [arendst/Tasmota#15483] [Q&A] [answered]
Hi,
I am relatively new and not very familiar with the tasmota firmware, but I have flashed an esp-01s with the firmware and it now works. 
Now I am faced with the problem that I would like to connect a sensor that has a digital out or an analogue out. To do this, I disconnected an ESP-01s DHT11 v1.0 board (with this sensor, a data query on gpio 2 with DHT11 is possible in the config) from the sensor and then connected it to the new sensor. Unfortunately, there is no way to declare this as an input on the gpio pin. And here is my problem - unfortunately I cannot find a configuration that reads the data from the sensor. With the DHT11 board, the gpio 2 pin is connected directly to the sensor and only the power management is taken over. Thank you very much.

**Comment by barbudor** [comment, marked as answer]
Esp01 is generally a bad choice as a development board.

GPIO 0, 1, 2 and 3 that are available on the edge connector are strictly digital. Esp01 does not provide access to the esp8266 only analog input. Unless you are ready for fine soldering, forget analog on esp01.

You don't say which sensor you are trying to connect to the esp so I can't help you much without details.
Beside my 1st comment, it is still possible to connect various sensors to the esp01 such as Ds18x20 or i2c sensors.

**Reply by tecbeat** [reply]
Ok this is not so nice. My Sensor is this on:
[Soil Moisture Sensor](https://smile.amazon.de/gp/product/B07PPSYJW8/ref=ppx_yo_dt_b_asin_title_o07_s00?ie=UTF8&psc=1
)

and I will try the i2c sensor the first I already tried.

**Reply by barbudor** [reply]
If by digital sensor you mean a binary sensor (on/off) in Tasmota vocabulary this is named a Switch.
So you should define the GPIO as Switch with index 1.
Switches are normally intended to drive Relay outputs (remember Tasmota original goal) but you can "decouple" the switch using different method explained in https://tasmota.github.io/docs/Buttons-and-Switches/
Choosing from the 3 possible methods will mostly depends on what you want to achieve, which you haven't elaborated on.

**Reply by barbudor** [reply]
So for the digital output see my answer above
For analog, no way on esp01
I would strongly advise to move to Nodemcu board.
Especially the newests based on ESP32-C3 (not the original ESP32 but the C3 variant) provides much more capabilities than the old Esp8266 (more analog input, more powerful processor, tasmota Berry scripting language...)

**Reply by barbudor** [reply]
And Nodemcu boards are far superior to mini-D1
They have a much better designed power supply section and works much more reliably.
The mini-D1 ESP32c3 is to be avoided at all cost as it has some design issues.

**Reply by tecbeat** [reply]
Many thanks!

**Reply by sfromis** [reply]
If you "need" analog with the bad ESP-01 board, you *could* also have an external ADC chip on I2C, and have any other sensors on I2C too. But I'd really recommend something better han ESP-01 which is something like the worst option.
